include/modmesh/SimpleArray.hpp

```diff
@@ -277,10 +360,10 @@ class SimpleArray
     }
 
     template < typename ... Args >
-    value_type const & operator()(Args ... args) const { return data(buffer_offset(m_stride, args...)); }
+    value_type const & operator()(Args ... args) const { return m_body[buffer_offset(m_stride, args...)]; }
```

Comment:
Use the body pointer to replace the original data pointer.  This implements the ghost index access.